src/functions/arrayDeep.js

```diff
@@ -0,0 +1,86 @@
+import {isSequence} from "../core/sequence";
+import {wrap} from "../core/wrap";
+
+import {array} from "./array";
+
+import {NotBoundedError} from "../errors/NotBoundedError";
+
+export const arrayDeep = wrap({
+    name: "arrayDeep",
+    summary: "Recursively acquire an in-memory array from a sequence potentially containing more sequences.",
+    docs: process.env.NODE_ENV !== "development" ? undefined : {
+        introduced: "higher@1.0.0",
+        expects: (`
+            The function expects a single known-bounded sequence as input.
+        `),
+        returns: (`
+            The function returns an array containing the elements of the input
+            sequence. Elements of that input sequence that were known-bounded
+            arrays also recursively have the operation applied to them, such
+            that a sequence of sequences becomes an array of arrays, and so on.
+            /Sequences which directly represent the contents of an in-memory
+            array will have that source array given as output.
+            To always acquire a new array object that is guaranteed to be safe
+            to modify, use @newArrayDeep.
+        `),
+        examples: [
+            "basicUsage",
+        ],
```

Comment:
`throws` deliberately omitted in anticipation of https://github.com/pineapplemachine/higher/issues/102